blocks/api/factory.js

```diff
@@ -67,6 +68,48 @@ export function createBlock( name, blockAttributes = {} ) {
 }
 
 /**
+ * Returns an array of possible block transformations that could happen on the set of blocks received as argument.
+ *
+ * @param  {Array}  blocks Blocks array
+ * @return {Array}         Array of possible block transformations
+ */
+export function getPossibleBlockTransformations( blocks ) {
+	if ( ! blocks || ! blocks[ 0 ] ) {
```

Comment:
Is this meant to be testing for a falsey or empty array? If so, I would suggest at least testing `Array#length` instead of accessing the first member directly. Alternatively consider [`_.isEmpty`](https://lodash.com/docs/4.17.4#isEmpty).

_Edit:_ Okay, I see we're working on the first entry in the set. Maybe we could assign `sourceBlock` as `first( blocks )` and test its truthiness here?

https://lodash.com/docs/4.17.4#first